--- gst-plugins-good-0.10.31/ext/taglib/gstapev2mux.cc	2011-12-30 13:59:08.000000000 +0000
+++ gst-plugins-good-0.10.31-new/ext/taglib/gstapev2mux.cc	2026-03-13 06:43:08.477712500 +0000
@@ -64,16 +64,15 @@
     GST_PAD_ALWAYS,
     GST_STATIC_CAPS ("application/x-apetag"));
 
-GST_BOILERPLATE (GstApev2Mux, gst_apev2_mux, GstTagLibMux,
-    GST_TYPE_TAG_LIB_MUX);
+G_DEFINE_TYPE(GstApev2Mux, gst_apev2_mux, GST_TYPE_TAG_LIB_MUX)
 
 static GstBuffer *gst_apev2_mux_render_tag (GstTagLibMux * mux,
     GstTagList * taglist);
 
 static void
-gst_apev2_mux_base_init (gpointer g_class)
+gst_apev2_mux_class_init (GstApev2MuxClass * klass)
 {
-  GstElementClass *element_class = GST_ELEMENT_CLASS (g_class);
+  GstElementClass *element_class = GST_ELEMENT_CLASS (klass);
 
   gst_element_class_add_static_pad_template (element_class, &src_template);
 
@@ -84,17 +83,13 @@
 
   GST_DEBUG_CATEGORY_INIT (gst_apev2_mux_debug, "apev2mux", 0,
       "taglib-based APEv2 tag muxer");
-}
 
-static void
-gst_apev2_mux_class_init (GstApev2MuxClass * klass)
-{
   GST_TAG_LIB_MUX_CLASS (klass)->render_tag =
       GST_DEBUG_FUNCPTR (gst_apev2_mux_render_tag);
 }
 
 static void
-gst_apev2_mux_init (GstApev2Mux * apev2mux, GstApev2MuxClass * apev2mux_class)
+gst_apev2_mux_init (GstApev2Mux *apev2mux)
 {
   /* nothing to do */
 }
--- gst-plugins-good-0.10.31/ext/taglib/gstid3v2mux.cc	2011-12-30 13:59:08.000000000 +0000
+++ gst-plugins-good-0.10.31-new/ext/taglib/gstid3v2mux.cc	2026-03-13 06:43:44.376986399 +0000
@@ -73,16 +73,15 @@
     GST_STATIC_CAPS ("application/x-id3"));
 
 
-GST_BOILERPLATE (GstId3v2Mux, gst_id3v2_mux, GstTagLibMux,
-    GST_TYPE_TAG_LIB_MUX);
+G_DEFINE_TYPE(GstId3v2Mux, gst_id3v2_mux, GST_TYPE_TAG_LIB_MUX)
 
 static GstBuffer *gst_id3v2_mux_render_tag (GstTagLibMux * mux,
     GstTagList * taglist);
 
 static void
-gst_id3v2_mux_base_init (gpointer g_class)
+gst_id3v2_mux_class_init (GstId3v2MuxClass * klass)
 {
-  GstElementClass *element_class = GST_ELEMENT_CLASS (g_class);
+  GstElementClass *element_class = GST_ELEMENT_CLASS (klass);
 
   gst_element_class_add_static_pad_template (element_class, &src_template);
 
@@ -93,17 +92,13 @@
 
   GST_DEBUG_CATEGORY_INIT (gst_id3v2_mux_debug, "id3v2mux", 0,
       "taglib-based ID3v2 tag muxer");
-}
 
-static void
-gst_id3v2_mux_class_init (GstId3v2MuxClass * klass)
-{
   GST_TAG_LIB_MUX_CLASS (klass)->render_tag =
       GST_DEBUG_FUNCPTR (gst_id3v2_mux_render_tag);
 }
 
 static void
-gst_id3v2_mux_init (GstId3v2Mux * id3v2mux, GstId3v2MuxClass * id3v2mux_class)
+gst_id3v2_mux_init (GstId3v2Mux * id3v2mux)
 {
   /* nothing to do */
 }
