--- gnome-utils-2.32.0/gnome-dictionary/libgdict/gdict-client-context.h	2026-03-10 11:15:42.937834287 +0000
+++ gnome-utils-2.32.0/gnome-dictionary/libgdict/gdict-client-context.h2	2026-03-10 11:15:32.928807085 +0000
@@ -21,6 +21,7 @@
 #define __GDICT_CLIENT_CONTEXT_H__
 
 #include <glib-object.h>
+#include "gdict-context.h"
 
 #define GDICT_TYPE_CLIENT_CONTEXT		(gdict_client_context_get_type ())
 #define GDICT_CLIENT_CONTEXT(obj)		(G_TYPE_CHECK_INSTANCE_CAST ((obj), GDICT_TYPE_CLIENT_CONTEXT, GdictClientContext))
